File: app/assets/javascripts/EvidenceVariable.js
const mongoose = require('mongoose');
const { DataElementSchema } = require('./basetypes/DataElement');
const Code = require('./basetypes/Code');
const Interval = require('./basetypes/Interval');
const Quantity = require('./basetypes/Quantity');
const DateTime = require('./basetypes/DateTime');
const { ComponentSchema } = require('./Component');
const { FacilityLocationSchema } = require('./FacilityLocation');
const { IdSchema } = require('./Id');
const Any = require('./basetypes/Any');

const [Number, String] = [
  mongoose.Schema.Types.Number,
  mongoose.Schema.Types.String,
];

const EvidenceVariableSchema = DataElementSchema({
  url: Any,
  identifier: Any,
  version: Any,
  name: Any,
  title: Any,
  shortTitle: Any,
  subtitle: Any,
  status: Any,
  date: Any,
  publisher: Any,
  contact: Any,
  description: Any,
  note: Any,
  useContext: Any,
  jurisdiction: Any,
  copyright: Any,
  approvalDate: Any,
  lastReviewDate: Any,
  effectivePeriod: Any,
  topic: Any,
  author: Any,
  editor: Any,
  reviewer: Any,
  endorser: Any,
  relatedArtifact: Any,
  type: Any,
  characteristic: Any,
  qdmVersion: { type: String, default: '4.0.0' },
  _type: { type: String, default: 'EvidenceVariable' },

});

module.exports.EvidenceVariableSchema = EvidenceVariableSchema;
module.exports.EvidenceVariable = mongoose.model('EvidenceVariable', EvidenceVariableSchema);

File: app/models/qdm/evidence_variable.rb
module QDM
  # app/models/qdm/evidence_variable.rb
  class EvidenceVariable < DataElement
    include Mongoid::Document
    embedded_in :patient
    field :url
    field :identifier
    field :version
    field :name
    field :title
    field :shortTitle
    field :subtitle
    field :status
    field :date
    field :publisher
    field :contact
    field :description
    field :note
    field :useContext
    field :jurisdiction
    field :copyright
    field :approvalDate
    field :lastReviewDate
    field :effectivePeriod
    field :topic
    field :author
    field :editor
    field :reviewer
    field :endorser
    field :relatedArtifact
    field :type
    field :characteristic
    field :qdmVersion, type: String, default: '4.0.0'
  end
end
